### PR TITLE
ref(TS) Improve TS

### PR DESCRIPTION
**Pull request by robertpin**
Remove unnecessary @ts-ignore
Convert some files to TS
Re-enable eslint lines-around-comment after import block

### CODE REVIEW

File: react/features/base/premeeting/components/web/Preview.tsx [hunk 1]
```diff
@@ -86,16 +87,16 @@ function Preview(props: Props) {
  * @param {Props} ownProps - The own props of the component.
  * @returns {Props}
  */
-function _mapStateToProps(state, ownProps) {
+function _mapStateToProps(state: IReduxState, ownProps: Partial<Props>) {
```
Comment (horymury) [inline react/features/base/premeeting/components/web/Preview.tsx:90]: could we go with useSelector approach instead?

File: react/features/base/participants/preloadImage.native.ts [hunk 2]
```diff
@@ -9,14 +6,16 @@ import { isIconUrl } from './functions';
  * Tries to preload an image.
  *
  * @param {string | Object} src - Source of the avatar.
+ * @param {boolean} _isUsingCORS - Used on web.
  * @returns {Promise}
  */
-export function preloadImage(src: string | Object): Promise<Object> {
+export function preloadImage(src: string | Object, _isUsingCORS: boolean): Promise<any> {
```
Comment (horymury) [inline react/features/base/participants/preloadImage.native.ts:12]: Seems that the web function has a `useCors`s and `tryOnce` param, and both native and web have the same payload for the promise resolve
```
{
  src: string,
  isUsingCORS: boolean
}
```

**Review by horymury — CHANGES_REQUESTED** [review]

File: react/features/base/premeeting/components/web/Preview.tsx (hunk 1 above)
Comment (robertpin) [reply react/features/base/premeeting/components/web/Preview.tsx:90]: We could, but that's not a TS change; it's a refactor for the component

File: react/features/base/participants/preloadImage.native.ts (hunk 2 above)
Comment (robertpin) [reply react/features/base/participants/preloadImage.native.ts:12]: Not really. The native also has a case of `Promise.resolve(src)`. So then the payload can be either like you wrote or just `string | Object`.

**Comment by robertpin** [comment]
Jenkins test this please

**Review by horymury — APPROVED** [review]

**Comment by robertpin** [comment]
Jenkins test this please

**Comment by robertpin** [comment]
Jenkins test this please

File: react/features/base/color-scheme/ColorSchemeRegistry.ts [hunk 3]
```diff
@@ -34,13 +33,13 @@ class ColorSchemeRegistry {
...
      *
-     * @param {Object | Function} stateful - An object or function that can be
+     * @param {IStateful} stateful - An object or function that can be
      * resolved to Redux state using the {@code toState} function.
      * @param {string} componentName - The name of the component whose style we
      * want to retrieve.
      * @returns {StyleType}
      */
-    get(stateful: Object | Function, componentName: string): StyleType {
+    get(stateful: IStateful, componentName: string): StyleType {
```
Comment (saghul) [inline react/features/base/color-scheme/ColorSchemeRegistry.ts:42]: This whole thing is legacy and will go away. I think it's best not to "modernize" it.

File: react/features/toolbox/components/web/DialogPortal.ts [hunk 4]
```diff
@@ -45,13 +43,13 @@ function DialogPortal({ children, className, style, getRef, setSize }: Props) {
...
     });
-    const timerRef = useRef();
+    const timerRef = useRef<number>();
 
     useEffect(() => {
         if (style) {
             for (const styleProp of Object.keys(style)) {
                 // https://github.com/facebook/flow/issues/3733
-                const objStyle: Object = portalTarget.style;
+                const objStyle: any = portalTarget.style;
```
Comment (saghul) [inline react/features/toolbox/components/web/DialogPortal.ts:52]: Drop the link above referring to the flow docs?

**Review by saghul — COMMENTED** [review]
Left a couple of minor comments

**Review by saghul — APPROVED** [review]

**Pull request merged**